cabal-install/Distribution/Client/Dependency/Modular/Message.hs

```diff
@@ -47,6 +47,9 @@ showMessages p sl = go [] 0
     go v l (TryF qfn b : Enter : Failure c fr : Leave : ms) = (atLevel (add (F qfn) v) l $ "rejecting: " ++ showQFNBool qfn b ++ showFR c fr) (go v l ms)
     go v l (TryS qsn b : Enter : Failure c fr : Leave : ms) = (atLevel (add (S qsn) v) l $ "rejecting: " ++ showQSNBool qsn b ++ showFR c fr) (go v l ms)
     go v l (Next (Goal (P qpn) gr) : TryP qpn' i : ms@(Enter : Next _ : _)) = (atLevel (add (P qpn) v) l $ "trying: " ++ showQPNPOpt qpn' i ++ showGRs gr) (go (add (P qpn) v) l ms)
+    go v l (Next (Goal (P qpn) gr) : Failure c fr : ms) =
+        let v' = add (P qpn) v
+        in (atLevel v' l $ showPackageGoal qpn gr) $ (atLevel v' l $ showFailure c fr) (go v l ms)
```

Comment:
Could you explain the idea here? If a package fails immediately, without any tries, then it already indicates that the package does not exist? And in such a case, you add the package to the filter, so that the message will always be shown? Wouldn't it be easier to just directly catch `BuildFailureNotInIndex` and always show that? Are there other failures that will now also be shown, and if so, is that desirable, or unexpected?